code/modules/projectiles/guns/projectile/launchers.dm

```diff
@@ -83,3 +83,752 @@
+							
+/obj/item/projectile/a66mm/Initialize()
+	..()
+	rocket_stop = list()
+	rocket_stop += /mob/living
+	rocket_stop += /obj/mecha
+	rocket_stop += /turf/unsimulated/wall
+	rocket_stop += /turf/simulated/wall
+	rocket_stop += /obj/machinery/door/poddoor
+	rocket_stop += /obj/machinery/porta_turret
+	rocket_stop += /obj/machinery/door/airlock/vault
+	rocket_stop += /obj/machinery/door/airlock/highsecurity
+	rocket_stop += /obj/structure/shuttle
+	rocket_stop += /obj/structure/girder
+	rocket_stop += /obj/structure/window/plasmabasic
+	rocket_stop += /obj/structure/window/full/plasmabasic
+	rocket_stop += /obj/structure/window/plasmareinforced
+	rocket_stop += /obj/structure/window/full/plasmareinforced
+	
+/obj/item/projectile/a66mm/fire()
+	var/spd_multi = 1 // Speed multiplayer for rocket travel.
+	var/xd = loc.x // X location of next tile to move to.
+	var/yd = loc.y // Y location of next tile to move to.
+	transform = turn(transform, Angle)
+	
+	while(loc)
+		xd += sin(Angle)*spd_multi
+		yd += cos(Angle)*spd_multi
+		if(round(xd, 1) <= 0 || round(xd,1) >= world.maxx)
+			on_range()
+			break
+		if(round(yd, 1) <= 0 || round(yd, 1) >= world.maxy)
+			on_range()
+			break
+		new /obj/effect/temp_visual/smoke_puff(loc)
+		var/turf/next_t = locate(round(xd, 1), round(yd, 1), z)
+		if(prehit(next_t))
+			loc = next_t
+			Range()
+		else
+			break
+		sleep(max(1, speed))
+		
+/obj/item/projectile/a66mm/prehit(turf/target as turf)
+	// Check for anything that stops the rocket.
+	for(var/i in rocket_stop)
+		if(istype(target, i))
+			on_hit(target)
+			qdel(src)
+			return FALSE
+		for(var/atom/j in target.contents)
+			if(istype(j, i) && j.density)
+				on_hit(target)
+				qdel(src)
+				return FALSE
+
+	// Anything that didn't stop the rocket, destroy it.
+	var/fx_hit_flag = FALSE
+	for(var/obj/j in target.contents)
+		if(src == j)
+			continue
+		var/skip_obj = FALSE
+		for(var/i in rocket_stop)
+			if(istype(j, i))
+				skip_obj = TRUE
+				break
+		if(skip_obj)
+			continue
+		if(istype(j, /obj/machinery/door))
+			j.take_damage(j.obj_integrity)
+			fx_hit_flag = TRUE
+		if(istype(j, /obj/machinery/door/airlock/multi_tile))
+			qdel(j)
+			fx_hit_flag = TRUE
+		else if(j.density)
+			j.take_damage(j.obj_integrity)
+			fx_hit_flag = TRUE
+			
+	if(fx_hit_flag)
+		sfx_hit(50, 33)
+		ffx_hit(target)
+
+	// Destroy structures or anything that was spawned from removal.
+	for(var/obj/j in target.contents)
+		var/skip_obj = FALSE
+		for(var/i in rocket_stop)
+			if(istype(j, i))
+				skip_obj = TRUE
+				break
+		if(skip_obj)
+			continue
+		if(istype(j, /obj/machinery/constructable_frame))
+			qdel(j)
+		else if(istype(j, /obj/structure/grille/broken))
+			continue
+		else if(istype(j, /obj/structure/table_frame))
+			continue
+		else if(istype(j, /obj/structure))
+			if(j.density)
+				qdel(j)
+
+	return TRUE
+	
+/obj/item/projectile/a66mm/on_hit(turf/target, blocked = 0)
+	if(locate(/obj/mecha) in target.contents)
```

Comment:
Bullets... that ignore antags?

Reply:
I know friendly fire is possible with other weapons but I didn’t want one antag to ruin another antag’s round with one miss clicked rocket.